src/NuGet.Core/NuGet.Build.Tasks.Pack/Pack.targets

```diff
@@ -139,6 +139,51 @@ Copyright (c) .NET Foundation. All rights reserved.
           TaskParameter="TargetOutputs"
           ItemName="_ProjectReferences" />
     </MSBuild>
+
+    <MSBuild
+      Condition="'$(PackBuildOutput)' == 'true'"
+      Projects="$(MSBuildProjectFullPath)"
+      Targets="BuiltProjectOutputGroup"
```

Comment:
yes, anything with CopyToOutputDirectory = true is included , but we filter with a hard coded list of extensions we allow in the nupkg .